app/uk/gov/hmrc/trackingconsentfrontend/assets/test/interfaces/optimizelyCommunicatorFactory.spec.ts

```diff
@@ -2,6 +2,43 @@ import optimizelyCommunicatorFactory from '../../src/interfaces/optimizelyCommun
 
 describe('sendPreferences', () => {
   let testScope
+
+  const getScripts = () => Array.from(document.getElementsByTagName('script'))
+  const getScriptSrcs = () => getScripts().map(e => e.getAttribute('src'))
+  const exampleTCScriptURL = '/tracking-consent/tracking.js'
+
+  const setupScriptWithID = (id, scriptSrc: string) => {
+    const originalScripts = getScripts()
+    const head = document.getElementsByTagName('head')[0]
+    const scriptTagWithAttr = document.createElement('script')
+    if (scriptSrc) {
+      scriptTagWithAttr.setAttribute('src', scriptSrc)
+    }
+    if (id) {
+      scriptTagWithAttr.setAttribute('data-optimizely-id', id)
+    }
+    head.appendChild(scriptTagWithAttr)
+
+    return {
+      trigger: () => {
```

Comment:
I would just have trigger as a separate function as it doesn't depend on the consts above and perhaps rename to something like `sendMeasurementAllowed()`